consensus/spos/bls/blsSubroundsFactory.go

```diff
@@ -107,7 +107,32 @@ func (fct *factory) GenerateSubrounds() error {
 	fct.consensusCore.Chronology().RemoveAllSubrounds()
 	fct.worker.RemoveAllReceivedMessagesCalls()
 
-	err := fct.generateStartRoundSubround()
+	// TODO: Wee need to have a components holder here which shall be injected
+	// Task: MX-14746
+	extraSignerHandler := fct.consensusCore.SigningHandler().ShallowClone()
+	startRoundExtraSignersHolder := NewSubRoundStartExtraSignersHolder()
+	startRoundExtraSigner, err := NewSovereignSubRoundStartOutGoingTxData(extraSignerHandler)
```

Comment:
this is the general blsSubroundFactory. this should not have anything regarding outGoiungTxData. or nothing related to Sovereign

Reply:
a new factory ?

Reply:
As stated in the task from the comment, there has already been a task and PR for this: https://github.com/multiversx/mx-chain-go/pull/5717

This will be a component injected here.
The sovereign one will contain these extra signer register, but the regular one will have no registered extra signer